QuantConnect.IBAutomater/IBAutomater.cs

```diff
@@ -250,9 +265,32 @@ public StartResult Start(bool waitForExit)
                 else
                 {
                     // wait for completion of IBGateway login and configuration
-                    var message = _ibAutomaterInitializeEvent.WaitOne(TimeSpan.FromSeconds(60))
-                        ? "IB Automater initialized."
-                        : "IB Automater initialization timeout.";
+                    string message;
+                    if (_ibAutomaterInitializeEvent.WaitOne(TimeSpan.FromSeconds(60)))
+                    {
+                        message = "IB Automater initialized.";
+                    }
+                    else
+                    {
+                        if (_twoFactorConfirmationPending)
+                        {
+                            // wait for completion of two-factor authentication
+                            if (!_ibAutomaterInitializeEvent.WaitOne(TimeSpan.FromMinutes(3)))
```

Comment:
nit - It actually ends up being 4 minutes with the previous 60 seconds, maybe this could potentially be a config setting